backend/map-resources/include/map-resources/resource-conversion-inl.h

```diff
@@ -402,6 +417,225 @@ bool convertDepthMapToPointCloud(
   return true;
 }
 
+template <typename PointCloudType>
+bool convertPointCloudToDepthMap(
+    const PointCloudType& point_cloud_C, const aslam::Camera& camera,
+    const bool use_openni_format, const bool create_range_image,
+    cv::Mat* depth_map, cv::Mat* image) {
+  CHECK(!(
+      camera.getType() == aslam::Camera::Type::kLidar3D && !create_range_image))
+      << "When projecting point clouds using the Camera3DLidar camera models, "
+      << "only range images (vs depth maps) are a maningful representation, "
+      << "since the points are not projected onto an image plange, but a "
+      << "cylinder.";
+  CHECK_NOTNULL(depth_map);
+
+  static constexpr bool kAlwaysParallelize = false;
+  const size_t num_threads = common::getNumHardwareThreads();
+
+  // Determine what data is available in the point cloud.
+  const size_t num_points = getPointCloudSize(point_cloud_C);
+  const bool has_color = hasColorInformation(point_cloud_C) && image != nullptr;
+  const bool has_intensity =
+      hasScalarInformation(point_cloud_C) && image != nullptr;
+
+  // Initialize depth map / range image depdening on format.
+  Eigen::Matrix3Xd points_3D = Eigen::Matrix3Xd::Zero(3, num_points);
+  Eigen::VectorXd depth_values = Eigen::VectorXd::Zero(num_points);
+  Eigen::Matrix2Xd projections_2D = Eigen::Matrix2Xd::Zero(2, num_points);
+  double max_valid_depth_value = 0.;
+  if (use_openni_format) {
+    *depth_map = cv::Mat(
+        camera.imageHeight(), camera.imageWidth(), CV_16UC1, cv::Scalar(0u));
+    max_valid_depth_value =
```

Comment:
Your max value might be 1 less than what you want in some cases due to rounding later on, but I assume that's fine.